src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +135,1023 @@ static DamageTable D_80A63CE0 = {
+            }
+            break;
+
+        case TEXT_STATE_CHOICE:
+            EnBal_TryPurchaseMap(this, play);
+            break;
+
+        case TEXT_STATE_5:
+            EnBal_HandleConversation(this, play);
+            break;
+
+        case TEXT_STATE_DONE:
+            if (Message_ShouldAdvance(play)) {
+                EnBal_SetupGroundIdle(this);
+            }
+            break;
+
+        case TEXT_STATE_10:
+            if (Message_ShouldAdvance(play) && (this->textId == 0x1D08)) {
+                this->forceEyesShut = false;
+                this->eyeTexIndex = TINGLE_EYETEX_OPEN;
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    if (this->textId == 0x1D07) {
+        if ((this->skelAnime.curFrame > 29.0f) && (this->skelAnime.curFrame < 33.0f)) {
+            if (Animation_OnFrame(&this->skelAnime, 30.0f)) {
+                Actor_PlaySfx(&this->picto.actor, NA_SE_EV_CHINCLE_SPELL_EFFECT);
+            }
+            EnBal_ThrowMagicSparkles(this, play);
+        }
+        if (Animation_OnFrame(&this->skelAnime, 35.0f)) {
+            this->forceEyesShut = false;
+            this->eyeTexIndex = TINGLE_EYETEX_OPEN;
+        }
+    }
+}
+
+void EnBal_UpdateShadow(EnBal* this) {
+    f32 homeY;
+    f32 worldY;
+    f32 shadowScale;
+
+    if ((this->actionFunc != EnBal_GroundIdle) && (this->actionFunc != EnBal_FloatIdle)) {
+        homeY = this->picto.actor.home.pos.y - this->picto.actor.floorHeight;
+        worldY = this->picto.actor.world.pos.y - this->picto.actor.floorHeight;
+        shadowScale = (homeY - worldY) / homeY;
+        shadowScale = CLAMP_MIN(shadowScale, 0.0f);
+        this->picto.actor.shape.shadowScale = 12.0f * shadowScale;
+        this->picto.actor.shape.shadowAlpha = 255.0f * shadowScale;
+    }
+}
+
+void EnBal_SetRecognizedPlayerForm(void) {
+    switch (gSaveContext.save.playerForm) {
+        case PLAYER_FORM_HUMAN:
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_LOW_BIT);
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_HIGH_BIT);
+            break;
+
+        case PLAYER_FORM_DEKU:
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_LOW_BIT);
+            break;
+
+        case PLAYER_FORM_GORON:
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_HIGH_BIT);
+            break;
+
+        case PLAYER_FORM_ZORA:
+            break;
+
+        default:
+            break;
+    }
+}
+
+PlayerTransformation EnBal_GetRecognizedPlayerForm(void) {
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_LOW_BIT)) {
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_HIGH_BIT)) {
+            return PLAYER_FORM_HUMAN;
+        }
+        return PLAYER_FORM_DEKU;
+    } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_HIGH_BIT)) {
+        return PLAYER_FORM_GORON;
+    }
+    return PLAYER_FORM_ZORA;
```

Comment:
```suggestion
    } 
    
    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_RECOGNIZED_PLAYER_FORM_HIGH_BIT)) {
        return PLAYER_FORM_GORON;
    }

    return PLAYER_FORM_ZORA;
```
I think this should still work and help space it out